add a script to merge records of multiple databases

joergsteffens opened the pull request:
After my approach to extend the webui to offer some access to other user records (#449) got rejected, I abandoned this approach. As a workaround, I created an external script to merge the record tables of multiple users together.

The timetagger-mergedatabases.py is intended to merge TimeTagger record tables of multiple user databases into one user database, replacing the existing records. Extends the records by user information.

This can be used as a workaround to create a unified view of the records of all or a group of users.
Currently I run this script periodically via cron.

Feel free to merge it, or reject it. As an external script, it is quite easy maintainable by itself.

almarklein commented:
Thanks! I guess it's time for a folder with user-contributed stuff. Let's create a folder `contrib` in the root of the repo, and move this script there.

joergsteffens commented:
Excellent. I'll adapt the PR soon and possible add another feature to the script.

joergsteffens commented:
Added some more features to the script, moved it to contrib and added to the package. Adding to the package has been done in a separate commit. Remove it, if you want.

almarklein commented on setup.py:
```diff
@@ -33,6 +33,7 @@
     package_data={
         f"timetagger.{x}": ["*"] for x in ["common", "images", "app", "pages"]
     },
+    scripts=['contrib/multiuser_tweaks/timetagger_multiuser_tweaks.py'],
```
```suggestion
    scripts=["contrib/multiuser_tweaks/timetagger_multiuser_tweaks.py"],
```

almarklein commented on contrib/multiuser_tweaks/timetagger_multiuser_tweaks.py:
```diff
@@ -0,0 +1,345 @@
+    TimeTagger uses a separate Sqlite database per user.
+    This tool directly accesses the databases and offers to manipulate them.
+    Especially it can merge the records of multiple users
+    into one user database, replacing the existing records. 
```
```suggestion
    into one user database, replacing the existing records.
```

almarklein commented:
Thanks! I just fixed some linting issues.
